### PR TITLE
Short circuiting bulk operations #258

### PR DESCRIPTION
**Pull request by AbfrmBlr**

### CODE REVIEW

File: core/src/test/java/org/ehcache/EhcacheBasicGetAllTest.java [hunk 1]
```diff
@@ -55,6 +54,7 @@
 import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.never;
```
Comment (chrisdennis) [inline core/src/test/java/org/ehcache/EhcacheBasicGetAllTest.java]: This import is a duplicate.

File: core/src/test/java/org/ehcache/EhcacheBasicGetAllTest.java [hunk 2]
```diff
@@ -132,60 +132,26 @@ public void testGetAllNullKey() throws Exception {
    * Tests {@link Ehcache#getAll(Set)} for
    * <ul>
    *    <li>empty request key set</li>
-   *    <li>{@link Store#bulkComputeIfAbsent} throws before accessing loader</li>
+   *    <li>{@link Store#bulkComputeIfAbsent} </li>
```
Comment (chrisdennis) [inline core/src/test/java/org/ehcache/EhcacheBasicGetAllTest.java]: This line can be deleted entirely, right?  The comment doesn't apply any more.

File: core/src/test/java/org/ehcache/EhcacheBasicGetAllTest.java [hunk 3]
```diff
@@ -132,60 +132,26 @@ public void testGetAllNullKey() throws Exception {
...
 
     final Map<String, String> actual = ehcache.getAll(Collections.<String>emptySet());
     assertThat(actual, is(notNullValue()));
     assertThat(actual.isEmpty(), is(true));
-
-    verify(this.store).bulkComputeIfAbsent(eq(Collections.<String>emptySet()), getAnyIterableFunction());
-    verify(this.spiedResilienceStrategy)
+    
+    verify(this.store, never()).bulkComputeIfAbsent(eq(Collections.<String>emptySet()), getAnyIterableFunction());
+    verify(this.spiedResilienceStrategy, never())
```
Comment (chrisdennis) [inline core/src/test/java/org/ehcache/EhcacheBasicGetAllTest.java]: Given the verifyZeroInteractions that you inserted aren't these individual verifications redundant?

File: core/src/test/java/org/ehcache/EhcacheBasicGetAllTest.java [hunk 4]
```diff
@@ -195,21 +161,17 @@ public void testGetAllEmptyRequestCacheAccessExceptionBeforeWithLoader() throws
    * Tests {@link Ehcache#getAll(Set)} for
    * <ul>
    *    <li>empty request key set</li>
-   *    <li>{@link Store#bulkComputeIfAbsent} throws before accessing loader</li>
+   *    <li>{@link Store#bulkComputeIfAbsent} </li>
```
Comment (chrisdennis) [inline core/src/test/java/org/ehcache/EhcacheBasicGetAllTest.java]: same deal... this comment is no longer applicable.

File: core/src/test/java/org/ehcache/EhcacheBasicGetAllTest.java [hunk 5]
```diff
@@ -217,16 +179,17 @@ public void testGetAllEmptyRequestCacheAccessExceptionBeforeLoaderAllFail() thro
...
     assertThat(actual, equalTo(expected));
 
-    final InOrder ordered = inOrder(this.loaderWriter, this.spiedResilienceStrategy);
-    verify(this.store).bulkComputeIfAbsent(eq(Collections.<String>emptySet()), getAnyIterableFunction());
-    ordered.verify(this.loaderWriter).loadAll(Collections.<String>emptySet());
-    ordered.verify(this.spiedResilienceStrategy)
-        .getAllFailure(eq(Collections.<String>emptySet()), eq(expected), any(CacheAccessException.class));
+    verify(this.store, never()).bulkComputeIfAbsent(eq(Collections.<String>emptySet()), getAnyIterableFunction());
+    verify(this.spiedResilienceStrategy, never())
+        .getAllFailure(eq(Collections.<String>emptySet()), any(CacheAccessException.class));
```
Comment (chrisdennis) [inline core/src/test/java/org/ehcache/EhcacheBasicGetAllTest.java]: same deal: these verifications are redundant.

File: core/src/test/java/org/ehcache/EhcacheBasicPutAllTest.java [hunk 6]
```diff
@@ -30,6 +29,7 @@
 import org.mockito.Captor;
 import org.mockito.InOrder;
 import org.mockito.Mock;
+import org.mockito.Mockito;
```
Comment (chrisdennis) [inline core/src/test/java/org/ehcache/EhcacheBasicPutAllTest.java]: unused

File: core/src/test/java/org/ehcache/EhcacheBasicPutAllTest.java [hunk 7]
```diff
@@ -191,37 +192,9 @@ public void testPutAllEmptyRequestNoWriter() throws Exception {
     final Ehcache<String, String> ehcache = this.getEhcache(null);
     ehcache.putAll(Collections.<String, String>emptyMap());
 
-    verify(this.store).bulkCompute(eq(Collections.<String>emptySet()), getAnyEntryIterableFunction());
+    verify(this.store, never()).bulkCompute(eq(Collections.<String>emptySet()), getAnyEntryIterableFunction());
```
Comment (chrisdennis) [inline core/src/test/java/org/ehcache/EhcacheBasicPutAllTest.java]: redundant

File: core/src/test/java/org/ehcache/EhcacheBasicPutAllTest.java [hunk 8]
```diff
@@ -246,104 +219,10 @@ public void testPutAllEmptyRequestWriterNoneFail() throws Exception {
     final Ehcache<String, String> ehcache = this.getEhcache(this.cacheLoaderWriter);
     ehcache.putAll(Collections.<String, String>emptyMap());
 
-    verify(this.store).bulkCompute(eq(Collections.<String>emptySet()), getAnyEntryIterableFunction());
+    verify(this.store, never()).bulkCompute(eq(Collections.<String>emptySet()), getAnyEntryIterableFunction());
```
Comment (chrisdennis) [inline core/src/test/java/org/ehcache/EhcacheBasicPutAllTest.java]: redundant

File: core/src/test/java/org/ehcache/EhcacheBasicRemoveAllTest.java [hunk 9]
```diff
@@ -149,38 +150,10 @@ public void testRemoveAllEmptyRequestNoWriter() throws Exception {
 
     final Ehcache<String, String> ehcache = this.getEhcache(null);
     ehcache.removeAll(Collections.<String>emptySet());
-
-    verify(this.store).bulkCompute(eq(Collections.<String>emptySet()), getAnyEntryIterableFunction());
+    
+    verify(this.store, never()).bulkCompute(eq(Collections.<String>emptySet()), getAnyEntryIterableFunction());
```
Comment (chrisdennis) [inline core/src/test/java/org/ehcache/EhcacheBasicRemoveAllTest.java]: redundant

File: core/src/test/java/org/ehcache/EhcacheBasicRemoveAllTest.java [hunk 10]
```diff
@@ -205,105 +178,12 @@ public void testRemoveAllEmptyRequestWriterNoneFail() throws Exception {
     final Ehcache<String, String> ehcache = this.getEhcache(this.cacheLoaderWriter);
     ehcache.removeAll(Collections.<String>emptySet());
 
-    verify(this.store).bulkCompute(eq(Collections.<String>emptySet()), getAnyEntryIterableFunction());
+    verify(this.store, never()).bulkCompute(eq(Collections.<String>emptySet()), getAnyEntryIterableFunction());
```
Comment (chrisdennis) [inline core/src/test/java/org/ehcache/EhcacheBasicRemoveAllTest.java]: redundant

**Comment by AbfrmBlr** [comment]
I have removed `verifyZeroInteractions` , and retained `never()` calls as it is more clear.

**Pull request merged**